rules/TypeDeclaration/Rector/Class_/ReturnTypeFromStrictTernaryRector.php

```diff
@@ -0,0 +1,172 @@
+use Rector\Core\ValueObject\PhpVersionFeature;
+use Rector\PHPStanStaticTypeMapper\Enum\TypeKind;
+use Rector\VendorLocker\ParentClassMethodTypeOverrideGuard;
+use Rector\VersionBonding\Contract\MinPhpVersionInterface;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Tests\TypeDeclaration\Rector\Class_\ReturnTypeFromStrictTernaryRector\ReturnTypeFromStrictTernaryRectorTest
+ */
+final class ReturnTypeFromStrictTernaryRector extends AbstractRector implements MinPhpVersionInterface
+{
+    public function __construct(
+        private readonly ParentClassMethodTypeOverrideGuard $parentClassMethodTypeOverrideGuard
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Add method return type based on strict ternary values', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+final class SomeClass
+{
+    public function getValue($number)
+    {
+        return $number ? 100 : 500;
+    }
+}
+CODE_SAMPLE
+
+                ,
+                <<<'CODE_SAMPLE'
+final class SomeClass
+{
+    public function getValue($number): int
+    {
+        return $number ? 100 : 500;
+    }
+}
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Class_::class];
+    }
+
+    /**
+     * @param Class_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $hasChanged = false;
+
+        foreach ($node->getMethods() as $classMethod) {
+            if ($classMethod->returnType instanceof Node) {
+                continue;
+            }
+
+            $onlyStmt = $classMethod->stmts[0] ?? null;
+            if (! $onlyStmt instanceof Return_) {
+                continue;
+            }
+
+            if (! $onlyStmt->expr instanceof Ternary) {
+                continue;
+            }
+
+            $ternary = $onlyStmt->expr;
+
+            // has scalar in if/else of ternary
+            if (! $this->isAlwaysScalarExpr($ternary->if)) {
+                continue;
+            }
+
+            if (! $this->isAlwaysScalarExpr($ternary->else)) {
+                continue;
+            }
+
+            /** @var Node\Expr $if */
+            $if = $ternary->if;
+
+            /** @var Node\Expr $else */
+            $else = $ternary->else;
+
+            $ifType = $this->staticTypeMapper->mapPhpParserNodePHPStanType($if);
+            $elseType = $this->staticTypeMapper->mapPhpParserNodePHPStanType($else);
+
+            if (! $this->areTypesEqual($ifType, $elseType)) {
+                continue;
+            }
+
+            $returnTypeNode = $this->staticTypeMapper->mapPHPStanTypeToPhpParserNode($ifType, TypeKind::RETURN);
+
+            if (! $this->parentClassMethodTypeOverrideGuard->shouldSkipReturnTypeChange($classMethod, $ifType)) {
```

Comment:
There is also `$this->classMethodReturnTypeOverrideGuard->shouldSkipClassMethod($node)` seems mostly  used on various rules

Reply:
Looking on other rules, the `$this->parentClassMethodTypeOverrideGuard->shouldSkipReturnTypeChange()` is used for "AddReturnTypeDeclaration" - which parent can has no type

For `$this->classMethodReturnTypeOverrideGuard->shouldSkipClassMethod($node)` seems better choice, as it also check magic method, chaotic class methods, already typed, has child return, or has class method return expr.

```suggestion
            if ($this->classMethodReturnTypeOverrideGuard->shouldSkipClassMethod($classMethod)) {
```

Reply:
I think we should have just one method to handle these, to avoid confusion

Reply:
Yes, both methods seems needs to be merged to cover all features.

Reply:
Could you handle that as well and add the check here with test?

Reply:
I will try.

Reply:
I'll just merge this one, so we have small iterations